src/main/scala/com/gu/liveappversions/S3Storage.scala

```diff
@@ -16,62 +17,71 @@ import scala.util.{ Failure, Success, Try }
 
 object S3Storage {
 
-  private val s3Client = AmazonS3ClientBuilder.standard()
-    .withCredentials(Aws.credentials("developerPlayground"))
-    .withRegion(Aws.euWest1.getName)
+  private val s3Client = S3Client
+    .builder()
+    .credentialsProvider(Aws.credentials("developerPlayground"))
+    .region(Aws.euWest1)
     .build()
 
   def fullKey(env: Env, partialKey: String): String = {
     val stagePrefix = if (env.stage == "PROD") "/" else s"/${env.stage}/"
     s"reserved-paths$stagePrefix$partialKey"
   }
 
-  def putJson(jsonToUpload: Json, env: Env, bucketName: String, partialKey: String): Try[PutObjectResult] = {
-
+  def putJson(jsonToUpload: Json, env: Env, bucketName: String, partialKey: String): Try[PutObjectResponse] = {
     val key = fullKey(env, partialKey)
+    val accessControl = if (env.stage == "PROD") ObjectCannedACL.PUBLIC_READ else ObjectCannedACL.PRIVATE
+    val bytes = jsonToUpload.toString().getBytes(StandardCharsets.UTF_8)
+    val stream = new ByteArrayInputStream(bytes)
+
+    val putRequest = PutObjectRequest.builder()
+      .bucket(bucketName)
+      .key(key)
+      .contentType("application/json")
+      .cacheControl("max-age=60")
+      .acl(accessControl)
+      .build()
 
-    val buildAttributesStream: ByteArrayInputStream = new ByteArrayInputStream(jsonToUpload.toString().getBytes)
-
-    val metadata = new ObjectMetadata()
-    metadata.setContentType("application/json")
-    metadata.setCacheControl("max-age=60")
-
-    val accessControl = if (env.stage == "PROD") {
-      CannedAccessControlList.PublicRead
-    } else {
-      CannedAccessControlList.Private //It's preferable to avoid serving test files via https://mobile.guardianapis.com/
-    }
-
-    val putObjectRequest = new PutObjectRequest(bucketName, key, buildAttributesStream, metadata)
-      .withCannedAcl(accessControl)
-
-    Try(s3Client.putObject(putObjectRequest)) match {
+    Try {
+      val requestBody = RequestBody.fromInputStream(stream, bytes.length)
+      s3Client.putObject(putRequest, requestBody)
+    } match {
       case Success(result) =>
         logger.info(s"Successfully uploaded new build info to S3 (bucket: $bucketName | key: $key)")
+        stream.close()
         Success(result)
       case Failure(exception) =>
         logger.error(s"Failed to uploaded build to S3 (bucket: $bucketName | key: $key) due to: $exception")
+        stream.close()
         Failure(exception)
     }
-
   }
 
   def getJsonString(env: Env, bucketName: String, partialKey: String): Try[Option[String]] = {
     val key = fullKey(env, partialKey)
+
+    val getRequest = GetObjectRequest.builder()
+      .bucket(bucketName)
+      .key(key)
+      .build()
+
     Try {
-      val s3Object = s3Client.getObject(bucketName, key)
-      Source.fromInputStream(s3Object.getObjectContent).mkString
+      val inputStream = s3Client.getObject(getRequest)
+      try {
+        Source.fromInputStream(inputStream).mkString
+      } finally {
```

Comment:
Just wondering if there is a more ideomatic scala why to incorporate stream closure here rather than having an inner try/finally block?

Reply:
Considering we are closing the stream both in the put and get methods, this might a place to abstract a small helper function that handles resource management and auto-closes the stream? I think there is something call the Using pattern and "try-with-resource" that might be useful here to handle the stream.

Reply:
Yeah, it's a bit confusing having the additional `try` here. If it's been introduced to the purposes of closing the stream I wouldn't worry as it's not necessary.